DNN Platform/Library/Security/Permissions/PermissionInfo.cs

```diff
@@ -1,50 +1,60 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
 namespace DotNetNuke.Security.Permissions
 {
     using System;
     using System.Data;
     using System.Xml.Serialization;
-
+
+    using DotNetNuke.Abstractions.Security.Permissions;
     using DotNetNuke.Common.Utilities;
-    using DotNetNuke.Entities;
-    using Newtonsoft.Json;
-
+    using DotNetNuke.Entities;
+    using Newtonsoft.Json;
+
     /// Project  : DotNetNuke
     /// Namespace: DotNetNuke.Security.Permissions
     /// Class    : PermissionInfo
     /// <summary>PermissionInfo provides the Entity Layer for Permissions.</summary>
     [Serializable]
-    public class PermissionInfo : BaseEntityInfo
+    public class PermissionInfo : BaseEntityInfo, IPermissionInfo
     {
-        /// <summary>Gets or sets the Mdoule Definition ID.</summary>
-        /// <returns>An Integer.</returns>
-        [XmlIgnore]
+        /// <inheritdoc cref="IPermissionInfo.ModuleDefId" />
+        [XmlIgnore]
         [JsonIgnore]
         public int ModuleDefID { get; set; }
 
-        /// <summary>Gets or sets the Permission Code.</summary>
-        /// <returns>A String.</returns>
+        /// <inheritdoc />
         [XmlElement("permissioncode")]
         public string PermissionCode { get; set; }
 
-        /// <summary>Gets or sets the Permission ID.</summary>
-        /// <returns>An Integer.</returns>
+        /// <inheritdoc cref="IPermissionInfo.PermissionID" />
         [XmlElement("permissionid")]
         public int PermissionID { get; set; }
 
-        /// <summary>Gets or sets the Permission Key.</summary>
-        /// <returns>A String.</returns>
+        /// <inheritdoc />
         [XmlElement("permissionkey")]
         public string PermissionKey { get; set; }
 
-        /// <summary>Gets or sets the Permission Name.</summary>
-        /// <returns>A String.</returns>
-        [XmlIgnore]
+        /// <inheritdoc />
+        [XmlIgnore]
         [JsonIgnore]
         public string PermissionName { get; set; }
 
+        /// <inheritdoc />
+        int IPermissionInfo.ModuleDefId
```

Comment:
Do we need to add `XmlIgnore` and `JsonIgnore`, like we have on the "real" property?

Reply:
Not sure how the JSON/XML-serializer handles implicit implemented interfaces, but better safe than sorry. I'll add them.